acceptance/suites/tests/authorization/default_rules.rb

```diff
@@ -0,0 +1,163 @@
+## These test the new trapperkeeper-authorization auth.conf default rules
+## as they're specified in FOSS puppet server.
+## The tests are written to assert the curl request was "not forbidden" rather
+## than expecting something meaningful from the endpoint. We're just trying to
+## test that authorization is allowing & rejecting requests as expected.
+##
+## The testing pattern is to call one of the two curl functions with a path,
+## and then one of the two assertion functions to validate allowed/denied.
+## The assertion functions look for regexes and status codes in the stdout
+## of the previous curl invocation.
+
+test_name 'Default auth.conf rules'
+
+step 'Turn on new auth support' do
+  modify_tk_config(
+    master,
+    File.join(master['puppetserver-confdir'], 'puppetserver.conf'),
+    {'jruby-puppet' => {'use-legacy-auth-conf' => false}})
+end
+
+teardown do
+  modify_tk_config(
+    master,
+    File.join(master['puppetserver-confdir'], 'puppetserver.conf'),
+    {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
+end
+
+def curl_authenticated(path)
+  curl = 'curl '
+  curl += '--cert $(puppet config print hostcert) '
+  curl += '--key $(puppet config print hostprivkey) '
+  curl += '--cacert $(puppet config print localcacert) '
+  curl += "--write-out '\\nSTATUSCODE=%{http_code}\\n' "
+  curl += 'https://$(hostname -f):8140'
+  curl += path
+  on(master, curl)
+end
+
+def curl_unauthenticated(path)
+  curl = 'curl --insecure '
+  curl += "--write-out '\\nSTATUSCODE=%{http_code}\\n' "
+  curl += 'https://$(hostname -f):8140'
+  curl += path
+  on(master, curl)
+end
+
+def assert_allowed(expected_statuscode = 200)
+  assert_no_match(/Forbidden request/, stdout)
+  assert_match(/^STATUSCODE=#{expected_statuscode}$/, stdout)
+end
+
+def assert_denied(expected_stdout)
+  assert_match(/Forbidden request/, stdout)
+  assert_match(expected_stdout, stdout)
+  assert_match(/^STATUSCODE=403$/, stdout)
+end
+
+def report_query(node)
+  curl = "/puppet/v3/report/#{node}?environment=production "
+  curl += '-X PUT -H "Content-Type: text/pson" '
+  curl += '--data "{\"host\":\"$(hostname -f)\", \"metrics\":{}, \"logs\":[], \"resource_statuses\":{}}"'
+end
+
+with_puppet_running_on(master, {}) do
+  step 'environments endpoint' do
+    curl_authenticated('/puppet/v3/environments')
+    assert_allowed
+
+    curl_unauthenticated('/puppet/v3/environments')
+    assert_denied(/denied by rule 'puppetlabs environments'/)
+  end
+
+  step 'catalog endpoint' do
+    curl_authenticated('/puppet/v3/catalog/$(hostname -f)?environment=production')
+    assert_allowed
+
+    curl_authenticated('/puppet/v3/catalog/notme?environment=production')
+    assert_denied(/denied by rule 'puppetlabs catalog'/)
+
+    curl_unauthenticated('/puppet/v3/catalog/$(hostname -f)?environment=production')
+    assert_denied(/denied by rule 'puppetlabs catalog'/)
```

Comment:
Might be good to add an example that does a catalog `post` to confirm that it can be successful / fail as expected - since that's what a real-world agent would do and `post` is explicitly part of the rule definition.